### PR TITLE
MAILBOX-339 Remove UDT from Cassandra primary keys

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/migration/MailboxPathV2MigrationTest.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+public class MailboxPathV2MigrationTest {
+
+    public static final MailboxPath MAILBOX_PATH_1 = MailboxPath.forUser("bob", "Important");
+    public static final int UID_VALIDITY_1 = 452;
+    public static final SimpleMailbox MAILBOX_1 = new SimpleMailbox(MAILBOX_PATH_1, UID_VALIDITY_1);
+    public static final CassandraId MAILBOX_ID_1 = CassandraId.timeBased();
+
+    static {
+        MAILBOX_1.setMailboxId(MAILBOX_ID_1);
```
Comment (mbaechler) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/migration/MailboxPathV2MigrationTest.java]: you should not need that, you have @BeforeClass for that

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/migration/MailboxPathV2MigrationTest.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.store.mail.model.impl.SimpleMailbox;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class MailboxPathV2MigrationTest {
+
+    public static final MailboxPath MAILBOX_PATH_1 = MailboxPath.forUser("bob", "Important");
```
Comment (mbaechler) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/migration/MailboxPathV2MigrationTest.java]: why public everywhere ?

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/migration/MailboxPathV2MigrationTest.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    @Test
+    public void newValuesShouldNotBeSavedInOldDAO() throws Exception {
+        mailboxMapper.save(MAILBOX_1);
+
+        assertThat(daoV1.retrieveId(MAILBOX_PATH_1).join())
+            .isEmpty();
+    }
+
+    @Test
+    public void readingOldValuesShouldNotMigrateThem() throws Exception {
```
Comment (mbaechler) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/migration/MailboxPathV2MigrationTest.java]: why didn't you choose that strategy ?

Comment (chibenwa) [reply mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/migration/MailboxPathV2MigrationTest.java]: Time of implementation.

If you want I can code it before the week-end :trollface:

Comment (mbaechler) [reply mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/migration/MailboxPathV2MigrationTest.java]: ok

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapperTest.java [hunk 4]
```diff
@@ -82,11 +95,132 @@ public void saveShouldNotRemoveOldMailboxPathWhenCreatingTheNewMailboxPathFails(
...
+        Mailbox childMailbox = new SimpleMailbox(childMailboxPath, UID_VALIDITY, childMailboxId);
+        mailboxDAO.save(childMailbox)
+            .join();
+        mailboxPathV2DAO.save(childMailboxPath, childMailboxId)
+            .join();
+    
+        boolean hasChildren = testee.hasChildren(MAILBOX, '.');
+    
+        assertThat(hasChildren).isTrue();
+    }
```
Comment (rouazana) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapperTest.java]: missing some "whenExistsInBothTable" tests, no?

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/utils/DriverExceptionHelperTest.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+import com.datastax.driver.core.exceptions.InvalidQueryException;
+
+public class DriverExceptionHelperTest {
+
+    @Test
+    public void handleStorageExceptionShouldPropagateWhenNoCause() {
+        CompletionException completionException = new CompletionException() {};
+
+        assertThatThrownBy(() ->
+            DriverExceptionHelper.handleStorageException(completionException))
```
Comment (rouazana) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/utils/DriverExceptionHelperTest.java]: indent (and everywhere in this file)

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/migration/MailboxPathV2MigrationTest.java (hunk 2 above)
Comment (chibenwa) [reply mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/migration/MailboxPathV2MigrationTest.java]: IDE bad configuration...

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/migration/MailboxPathV2MigrationTest.java (hunk 1 above)
Comment (chibenwa) [reply mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/migration/MailboxPathV2MigrationTest.java]: Smart...

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java [hunk 6]
```diff
@@ -108,12 +108,28 @@ public Mailbox findMailboxByPath(MailboxPath path) throws MailboxException {
...
+    private CompletableFuture<Optional<SimpleMailbox>> migrate(SimpleMailbox mailbox) {
+        CassandraId mailboxId = (CassandraId) mailbox.getMailboxId();
+        return mailboxPathV2DAO.save(mailbox.generateAssociatedPath(), mailboxId)
+            .thenCompose(success -> {
+                if (success) {
+                    mailboxPathDAO.delete(mailbox.generateAssociatedPath());
+                }
+                LOGGER.info("Concurrent execution lead to data race while migrating {} to 'mailboxPathV2DAO'.",
+                    mailbox.generateAssociatedPath());
+                return CompletableFuture.completedFuture(null);
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java]: Extract what to a function, please?

**Comment by mbaechler** [comment]
merged

**Pull request closed without merging**